tests/test_dataIdContainer.py

```diff
@@ -24,15 +24,22 @@
 import unittest
 import unittest.mock
 
-import lsst.daf.persistence
-import lsst.pipe.base as pipeBase
+try:
+    import lsst.daf.persistence as dafPersistence
+except ImportError:
+    dafPersistence = None
+try:
+    from lsst.pipe.base import DataIdContainer
+except ImportError:
+    DataIdContainer = None
 import lsst.utils.tests
 
 
+@unittest.skipIf(DataIdContainer is None, "Gen2 infrastructure is not available")
 class DataIdContainerTestCase(lsst.utils.tests.TestCase):
     def setUp(self):
-        self.container = pipeBase.DataIdContainer()
-        self.butler = unittest.mock.MagicMock(spec=lsst.daf.persistence.Butler)
+        self.container = DataIdContainer()
+        self.butler = unittest.mock.MagicMock(spec=dafPersistence.Butler)
```

Comment:
Wont this also fail if dafPersistence is None?

Reply:
yes, but the test is skipped because the gen2 code can't be imported because dafPersistence is missing. I can add an explicit `or` to the skipIf if you like.